### ISSUE TITLE
sys-net 'Loop in network not supported' error

### ISSUE THREAD

**Issue by Seabass5701** [QubesOS/qubes-issues#4059]
Last night I installed the latest version of Qubes on my laptop via usb disk drive write method. After installing, towards the end when it was configuring network settings, it showed an error that said something like PCI couldn't reset or something like that, and the PCI's that were causing the issue were: 0000:02:00.1 & 0000:02:00.0. I went ahead and finished the installation anyways. After, when I logged into my account on Qubes OS, it worked. However, upon logging in, when sys-net started, I opened up the Qubes settings for that VM, and tried to set up the networking settings to sys-firewall (default), but when I clicked apply, it gave me an error saying 'loop in network not supported'. I haven't found this problem anywhere else online except for on a google forum site, and their question remained unanswered. Therefore, I am stuck as to what I should do next, since I am new to all of this, and if somebody could lead me step by step into doing what I need to do to get this problem resolved, it would be greatly appreciated. Thanks.

**Comment by donob4n** [comment]
This is not a bug. sys-firewall has sys-net as net VM, if you try to set sys-net to sys-firewall you are making a loop (sys-net -> sys-firewall -> sys-net -> sys-firewall .....)

You should try to fix the problem with PCI devices. Try to attach to sys-net something like 'Network controller' or 'Ethernet controller' (in Qube Manager, settings, Devices tab). If you have problems with this open a new bug or try with qubes-user mail list adding information about your hardware.

**Comment by andrewdavidwong** [comment]
> Last night I installed the latest version of Qubes on my laptop via usb disk drive write method. After installing, towards the end when it was configuring network settings, it showed an error that said something like PCI couldn't reset or something like that, and the PCI's that were causing the issue were: 0000:02:00.1 & 0000:02:00.0. 

Could be a PCI strict-reset issue. See:
- https://www.qubes-os.org/doc/assigning-devices/
- https://www.qubes-os.org/faq/#i-created-a-usbvm-and-assigned-usb-controllers-to-it-now-the-usbvm-wont-boot

> I went ahead and finished the installation anyways. After, when I logged into my account on Qubes OS, it worked. However, upon logging in, when sys-net started, I opened up the Qubes settings for that VM, and tried to set up the networking settings to sys-firewall (default), but when I clicked apply, it gave me an error saying 'loop in network not supported'. I haven't found this problem anywhere else online except for on a google forum site, and their question remained unanswered. Therefore, I am stuck as to what I should do next, since I am new to all of this, and if somebody could lead me step by step into doing what I need to do to get this problem resolved, it would be greatly appreciated. Thanks.

As @donob4n explained, you are trying to arrange the VMs in such a way that they form a circular loop of network dependency, which doesn't make sense. If `sys-firewall` gets it network access from `sys-net`, and `sys-net` gets its network access from `sys-firewall`, then neither of them will have any useful network access. So, it's not an error. Qubes is just telling you that you're trying to do something that doesn't make sense (and therefore isn't supported). If `sys-net` was already working fine, then there's no reason to mess with it.

**Comment by Seabass5701** [comment]
Hello guys, just wanted to let you know that I solved the issue. I guess when I first installed Qubes, I installed it on a 32GB MicroSD card which was insufficient space for everything to be contained within it, so I instead created a new partition on my SSD with about 50GB, and once I installed Qubes on this partition, almost every problem I have been having went away. As for the network loop, I had no idea what was going on, and I appreciate you guys explaining it to me. I guess once I had first installed Qubes, I messed with the Qubes settings inside the sys-firewall and sys-net VMs, and didn't really know what I was doing, but once I reinstalled it and the settings were automatically configured, I was able to connect to the internet. I appreciate the help and responses, and if I encounter any other errors, I will inform the community of them, and, if need be, will request help from you devs. Much thanks again.
